--- a/libavcodec/Makefile	2015-05-31 10:55:01.494795804 +0200
+++ b/libavcodec/Makefile	2015-05-31 10:55:16.462530460 +0200
@@ -4,10 +4,6 @@
 
 HEADERS = avcodec.h                                                     \
           avfft.h                                                       \
-          rpi_qpu.h                                                     \
-          rpi_shader.h                                                  \
-          rpi_mailbox.h                                                 \
-          rpi_hevc_transform.h                                          \
           dv_profile.h                                                  \
           dxva2.h                                                       \
           old_codec_ids.h                                               \
@@ -37,9 +33,6 @@
        resample.o                                                       \
        resample2.o                                                      \
        utils.o                                                          \
-       rpi_qpu.o                                                        \
-       rpi_shader.o                                                     \
-       rpi_mailbox.o                                                    \
        vorbis_parser.o                                                  \
        xiph.o                                                           \
 
--- a/libavcodec/hevc.h	2015-05-31 10:29:34.000000000 +0200
+++ b/libavcodec/hevc.h	2015-05-31 11:04:26.486341721 +0200
@@ -37,7 +37,7 @@
 #include "videodsp.h"
 
 // define RPI to split the CABAC/prediction/transform into separate stages
-#define RPI
+//#define RPI
 
 #ifdef RPI
 
--- a/libavcodec/utils.c	2015-05-31 10:29:34.000000000 +0200
+++ b/libavcodec/utils.c	2015-05-31 11:13:05.543108709 +0200
@@ -26,7 +26,7 @@
  */
  
 // Move video buffers to GPU memory
-#define RPI_GPU_BUFFERS
+//#define RPI_GPU_BUFFERS
 
 #include "config.h"
 #include "libavutil/atomic.h"
--- a/libavcodec/rpi_qpu.c	2015-06-14 00:13:13.000000000 +0200
+++ b/libavcodec/rpi_qpu.c	2015-06-14 00:44:50.203459910 +0200
@@ -4,7 +4,7 @@
 // define RPI_TIME_TOTAL_VPU to print out how much time is spent in the VPI code
 //#define RPI_TIME_TOTAL_VPU
 // define RPI_ASYNC to run the VPU in a separate thread, need to make a separate call to check for completion
-#define RPI_ASYNC
+//#define RPI_ASYNC
 
 #include <stdio.h>
 #include <stdlib.h>
